### PR TITLE
Remove mkdirp and rimraf in favor of Node.js builtins

### PR DESCRIPTION
**Pull request by wojtekmaj**
In Node.js v10.12.0, `recursive` option was added to [`fs.mkdirSync`](https://nodejs.org/api/fs.html#fsmkdirsyncpath-options). This option allows you to create a directory and all its parent directories if they do not exist.

In Node.js v14.14.0, [`fs.rmSync`](https://nodejs.org/api/fs.html#fsrmsyncpath-options) was added. It's a built-in module that removes files and directories. The `recursive` option is used to remove directories and their contents.

Since pdf.js repository requires Node.js 18 and up, it's safe to assume `mkdirp` and `rimraf` can be safely replaced, thus reducing the number of dependencies in this project.

### CODE REVIEW

File: gulpfile.mjs [hunk 1]
```diff
@@ -2184,13 +2183,13 @@ gulp.task(
...
-      rimraf.sync(DIST_DIR);
-      mkdirp.sync(DIST_DIR);
+      fs.rmSync(DIST_DIR, { recursive: true, force: true });
+      fs.mkdirSync(DIST_DIR, { recursive: true });
       safeSpawnSync("git", ["clone", "--depth", "1", DIST_REPO_URL, DIST_DIR]);
 
       console.log();
       console.log("### Overwriting all files");
-      rimraf.sync(path.join(DIST_DIR, "*"));
+      fs.rmSync(DIST_DIR, { recursive: true, force: true });
```
Comment (wojtekmaj) [inline gulpfile.mjs:2192]: That's the only line I'm not sure of

Comment (Snuffleupagus) [reply gulpfile.mjs:2192]: If you run `gulp dist-pre` locally and compare the result with/without this patch you'll see that this change breaks things, since the output is no longer identical.

Hence this patch would *reintroduce* issue #15975, note in particular https://github.com/mozilla/pdf.js/issues/15975#issuecomment-1445400924.

File: gulpfile.mjs [hunk 2]
```diff
@@ -2063,14 +2061,15 @@ gulp.task("importl10n", async function () {
   if (!fs.existsSync(L10N_DIR)) {
     fs.mkdirSync(L10N_DIR);
   }
+
```
Comment (Snuffleupagus) [inline gulpfile.mjs:2064]: Please don't make unrelated changes.

**Review by Snuffleupagus — CHANGES_REQUESTED** [review]

File: gulpfile.mjs (hunk 2 above)
Comment (wojtekmaj) [reply gulpfile.mjs:2064]: Fixed

File: gulpfile.mjs (hunk 1 above)
Comment (wojtekmaj) [reply gulpfile.mjs]: Hmmm, trying something else. I don't see the difference in outputs now, but I need a fresh brain to be sure.

**Comment by Snuffleupagus** [comment]
It's probably a good idea to split this into two *separate* PRs, one for `mkdirp` and one for `rimraf`, since replacing `mkdirp` seems less problematical/risky and thus might be able to land sooner.

**Comment by calixteman** [comment]
Would it be possible to have few tests to check that we aren't breaking something again ?

**Comment by Snuffleupagus** [comment]
Closing this for now, since there's not been any updates for a couple of weeks and this cannot land in its current state (needs careful testing and no fixup commits for a start).
